### PR TITLE
[R2R] Add recipe for tabula-py

### LINKED ISSUE #5929: Package tabula
[`tabula-py`](https://github.com/chezou/tabula-py) is a simple Python wrapper of tabula-java, which can read table of PDF. You can read tables from PDF and convert into pandas's DataFrame. tabula-py also enables you to convert a PDF file into CSV/TSV/JSON file.

### PR DESCRIPTION
**Pull request by CurtLH**
Closes #5929

### CODE REVIEW

**Comment by CurtLH** [comment]
The [README](https://github.com/chezou/tabula-py#os) states the following about system compatibility:

> I confirmed working on macOS and Ubuntu. But some people confirm it works on Windows 10. See also following the setting procedure.

File: recipes/tabula/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  skip: True  # [win]
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - setuptools
```
Comment (sodre) [inline recipes/tabula/meta.yaml]: this is strange as a runtime requirement...

**Comment by CurtLH** [comment]
Hey @sodre, that sounds good to me!

**Comment by jakirkham** [comment]
Since you are looking at packaging Java libraries, it would be good if you could weigh in on the discussion about Java packages in issue ( https://github.com/conda-forge/conda-forge.github.io/issues/590 ).

File: recipes/tabula/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  script:
+    - rm tabula/*.jar  # [unix]
+    - del tabula\*.jar  # [win]
+    - python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  host:
+    - python
+    - pip
+    - flake8
```
Comment (sodre) [inline recipes/tabula/meta.yaml]: tabula-py does not build without flake8 dependency here.

File: recipes/tabula/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+test:
+  imports:
+    - tabula
+
+about:
+  home: https://github.com/chezou/tabula-py
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
```
Comment (sodre) [inline recipes/tabula/meta.yaml]: @CurtLH, can you submit a PR upstream to have the license file added? In the meantime it is packaged under the recipe folder.

Comment (CurtLH) [reply recipes/tabula/meta.yaml]: I just created a [PR](https://github.com/chezou/tabula-py/pull/94) to include the license in the source tarball.

**Comment by sodre** [comment]
@jakirkham, are you okay with the current structure for java-tabula?

**Comment by sodre** [comment]
Since I did quite a few commits, a second pair of eyes would be helpful.

@conda-forge/staged-recipes this is ready for review

**Comment by jakirkham** [comment]
> .are you okay with the current structure for java-tabula?

Far from an expert here. You've probably put more thought in this than I. Where do other package management systems place JARs normally?

**Comment by sodre** [comment]
In view of my review in #590, I think in the *short term* we should not patch upstream code to look for jars outside site-packages. 

Time permitting, I will engage with upstream and propose building the .jar dependency from source as was done with dask/knit.

@CurtLH, I apologize for the delay in getting this merged.

**Pull request merged**